src/discord-api.ts

```diff
@@ -141,3 +163,70 @@ const fetchData = async (url: string, token: string) =>
 			Accept: 'application/json',
 		},
 	})
+
+export const fetchAttachment = async (ctx: Context) => {
+	const request = ctx.req
+	const env = ctx.env
+
+	const attachment_url = new URL(ctx.req.param('url'))
+
+	console.log('attachment_url', attachment_url)
+
+	const params = new URLSearchParams(attachment_url.search)
+	if (params.get('ex') && params.get('is') && params.get('hm')) {
+		const expires = new Date(parseInt(params.get('ex') ?? '', 16) * 1000)
+		if (expires.getTime() > Date.now()) {
+			return redirectResponse(request, attachment_url.href, expires, 'original')
```

Comment:
It seems like you're putting in a full url as a path parameter atm? Like this: `https://discord-proxy.toddle.cloudflare.dev/api/https://discord.co/attachmentX.png`
That seems a bit strange (to me), perhaps you could use a pattern like this instead and always inject the origin server-side?
`https://discord-proxy.toddle.cloudflare.dev/api/attachmentX.png`

Reply:
We already have the whole url in the db so I'm not sure if it make sense to remove the origin from it and then later added it again 🤔